assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js

```diff
@@ -186,6 +186,44 @@ describe( 'AdBlockingRecoverySetupCTAWidget - SetupMain', () => {
 				dashboardURL
 			);
 		} );
+
+		it( 'should return to AdSense settings page when `Cancel` button is clicked', async () => {
+			const originalReferrer = document.referrer;
+
+			const settingsURL = registry
+				.select( CORE_SITE )
+				.getAdminURL( 'googlesitekit-settings' );
+
+			// Mock the document.referrer to be the AdSense settings page.
+			Object.defineProperty( document, 'referrer', {
+				writable: true,
+				value: settingsURL,
+			} );
+
+			const adSenseSettingsURL = `${ settingsURL }#/connected-services/adsense`;
+
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click( getByRole( 'button', { name: /cancel/i } ) );
+			} );
+
```

Comment:
We should probably duplicate the `mockTrackEvent` assertion as it's a separate but similar test and we'd probably be sure that this was not different. I don't feel super strongly about it, though.

Reply:
I'm already asserting the `mockTrackEvent` for the new tests. Please check the code below the `fireEvent.click()`. See: https://github.com/google/site-kit-wp/blob/a282efcd89b3e99e5457325984e180adbc2d06eb/assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js#L210-L214

If this differs from what you intended, could you please clarify it?

Reply:
Thank you for clarifying, I was under the impression that there was different events for this case, but that is not the case.